decode-error/src/lib.rs

```diff
@@ -21,6 +21,7 @@ use num_traits::FromPrimitive;
 /// [`ProgramError`]: https://docs.rs/solana-program-error/latest/solana_program_error/enum.ProgramError.html
 /// [`ProgramError::Custom`]: https://docs.rs/solana-program-error/latest/solana_program_error/enum.ProgramError.html#variant.Custom
 /// [`ToPrimitive`]: num_traits::ToPrimitive
+#[deprecated(since = "2.3.0", note = "Use `num_traits::FromPrimitive` instead")]
```

Comment:
Do we use specific functionalities from `num_traits::FromPrimitive`? For the changes in #12 we are using `TryFrom<u32>` and removed the `num_traits` dependency.

Reply:
Probably not, but we're also removing this crate very soon, so I wanted to go with the simplest option

Reply:
iirc we use this trait for some dumb explicit i32 <-> enum variant marshaling that's probably unnecessary (blame the phone guys)